src/Form/MessageSettingsForm.php

```diff
@@ -100,21 +100,17 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       ],
     ];
 
-    $form['purge']['purge_quota'] = [
-      '#type' => 'textfield',
-      '#title' => t('Messages quota'),
-      '#description' => t('Maximal (approximate) amount of messages.'),
-      '#default_value' => $config->get('purge_quota'),
-      '#states' => $states,
-    ];
+    /** @var \Drupal\message\MessagePurgeInterface $plugin */
+    foreach (array_keys(\Drupal::service('plugin.manager.message.purge')->getDefinitions()) as $plugin_id) {
+      $plugin = \Drupal::service('plugin.manager.message.purge')->createInstance($plugin_id);
+      $element = $plugin->configurationForm($form, $form_state);
+      $element['#states'] = $states;
 
-    $form['purge']['purge_days'] = [
-      '#type' => 'textfield',
-      '#title' => t('Purge messages older than'),
-      '#description' => t('Maximal message age in days.'),
-      '#default_value' => $config->get('purge_quota'),
-      '#states' => $states,
-    ];
+      // @todo: Set the default value.
+      // $element['#default_value'] = $config->get($plugin_id);
```

Comment:
@damiankloip so my plugin's config question is this. Given I have this settings yaml:

```yaml
# message/config/install/message.settings.yml
purge:
  quota: 0
  days: 0
```

I could set the default value by doing something like `$element['#default_value'] = $config->get('purge.' . $plugin_id);`

However, how would that work if the plugin is defined by another module? -- i.e. it's value isn't in Message's own settings yml file.

Am I taking the right direction here?